### PR TITLE
INTMDB-734: Add examples for golang CDK

### PR DESCRIPTION
**Pull request by maastha**
## Proposed changes

_Jira ticket:_ [INTMDB-734](https://jira.mongodb.org/browse/INTMDB-734)

Add examples for golang CDK.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [X] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code
- [ ] I have tested the CDK constructor in a CFN stack. See [TESTING.md](../TESTING.md)

### CODE REVIEW

**Comment by maastha** [comment]
Please let me know if project structure is fine. Planning to add for other languages as well.

**Comment by andreaangiolillo** [comment]
FYI: We plan to add examples for all the languages for max 3 CDK constructors. We will then create a DOCK ticket so that the doc team can add the missing tests. 
(We already have examples in Typescripts for all the CDK constructors)

File: examples/golang/atlas-basic.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+			AdvancedRegionConfigs: regionConfigs,
+			NumShards:             jsii.Number(1),
+		},
+	}
+
+	ipAccessListProps := mdb.IpAccessListProps{
+		AccessList: &[]*mdb.AccessListDefinition{
+			{
+				Comment:   jsii.String("Open comment"),
+				IpAddress: jsii.String("10.10.0.0/24"),
```
Comment (colm-quinn) [inline examples/golang/atlas-basic.go:45]: [nit] Can we replace the value with a placeholder string here? I can see people just using the code directly with no changes.

Ideally, lets define them in a struct passed into the method, so users can easily define in a single place.

Comment (colm-quinn) [reply examples/golang/atlas-basic.go:45]: See https://github.com/mongodb/awscdk-resources-mongodbatlas/blob/main/examples/l1-resources/cluster.ts for an example

**Review by colm-quinn — COMMENTED** [review]
I think we should encourage best practice. or at least better practice, for the magic strings, using [cdk runtime context](https://docs.aws.amazon.com/cdk/v2/guide/context.html) in our examples. 

That way, even if a user runs the code with no changes, they get prompted or an error to set the variables.

File: examples/golang/atlas-basic.go (hunk 1 above)
Comment (maastha) [reply examples/golang/atlas-basic.go]: updated examples

**Review by andreaangiolillo — APPROVED** [review]

**Pull request merged**